ducktape/cluster/json.py

```diff
@@ -31,7 +32,7 @@ class JsonCluster(Cluster):
     """An implementation of Cluster that uses static settings specified in a cluster file or json-serializeable dict
     """
 
-    def __init__(self, cluster_json=None, *args, **kwargs):
+    def __init__(self, cluster_json: Optional[Union[dict, str]]=None, *args, **kwargs):
```

Comment:
afaict cluster_json should never be a string. did you mean `Optional[dict] = None` ?

Reply:
I think I confused it with `cluster_file`. I will go back and check. Side note: ducktape needs some improvements around readability and type safety, and this pattern of `*args, **kwargs` and especially `for k,v   in kwargs.item(): setattr(self, k, f)` is high on the list. It is like dynamic typing + dynamic binding--on the opposite end of the spectrum from static typing.